Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -535,6 +535,22 @@ public final class SwiftTargetBuildDescription {
             args += ["-color-diagnostics"]
         }
 
+        // If this is a generated test discovery target, it might import a test
+        // target that is built with C++ interop enabled. In that case, the test
+        // discovery target must enable C++ interop as well
+        switch testTargetRole {
+        case .discovery:
+            for dependency in try self.target.recursiveTargetDependencies() {
```

Comment:
This is somewhat ugly, but it should go away once we have a better story for mixing C-interop and C++-interop in a single project.